### PR TITLE
Add gulp and lint

### PR DESCRIPTION
**Pull request by itsananderson**
So... this kinda evolved as I went. I actually ended up switching to ESLint. I think I have a reasonable set of defaults, but I'd welcome feedback on any rules we should turn on.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -2,28 +2,21 @@
...
-    return gulp.src(testSources).pipe(jslint({
-        options: {
-            node: true,
-            this: true
-        },
-        globals: ["describe", "it", "before", "after", "beforeEach", "afterEach"],
-        noFail: true
-    }));
+    return gulp.src(testSources)
+        .pipe(eslint())
```
Comment (benquarmby) [inline gulpfile.js]: Not really important, but the lint tasks could be combined now that there are no differences in the configuration.

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        ],
+        "quotes": [
+            "error",
+            "double"
+        ],
+        "semi": [
+            "error",
+            "always"
+        ]
+    }
```
Comment (benquarmby) [inline .eslintrc.json]: I would also consider adding `curly` and set it to `all`. I've seen brace-free blocks go bad too many times!

**Comment by benquarmby** [comment]
This is awesome :+1:

ESLint also plays very nicely with Visual Studio code.

**Comment by itsananderson** [comment]
Added curly checking and cleaned up the Gulp tasks. I'll go ahead and merge this

**Pull request merged**